server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -443,6 +445,22 @@ public ClusterState execute(ClusterState currentState) {
                             if (metadata.persistentSettings() != null) {
                                 Settings settings = metadata.persistentSettings();
                                 clusterSettings.validateUpdate(settings);
```

Comment:
Hopefully this never causes an issue, but I think this line should go below the modification of the `settings` below to ensure the updated `settings` are valid.

Reply:
A good catch, thanks! It may not cause real issue like you said, nevertheless it is better to be safe.